[cmake] Qt6 cmake errors

<!--- WARNING: This is an issue tracker. Before opening a new issue make sure you read https://github.com/PointCloudLibrary/pcl/blob/master/CONTRIBUTING.md#using-the-issue-tracker. -->
**Describe the bug**

When building with Qt6 on Ubuntu 24.04, cmake produces the following error:
```
 CMake Error at /usr/lib/x86_64-linux-gnu/cmake/Qt6Core/Qt6CoreVersionlessTargets.cmake:42 (message):
   Some (but not all) targets in this export set were already defined.

   Targets Defined: Qt::Core

   Targets not yet defined: Qt::CorePrivate

 Call Stack (most recent call first):
   /usr/lib/x86_64-linux-gnu/cmake/Qt6Core/Qt6CoreConfig.cmake:65 (include)
   /usr/share/cmake-3.28/Modules/CMakeFindDependencyMacro.cmake:76 (find_package)
   /usr/lib/x86_64-linux-gnu/cmake/Qt6/QtPublicDependencyHelpers.cmake:108 (find_dependency)
   /usr/lib/x86_64-linux-gnu/cmake/Qt6Concurrent/Qt6ConcurrentDependencies.cmake:39 (_qt_internal_find_qt_dependencies)
   /usr/lib/x86_64-linux-gnu/cmake/Qt6Concurrent/Qt6ConcurrentConfig.cmake:50 (include)
   /usr/lib/x86_64-linux-gnu/cmake/Qt6/Qt6Config.cmake:167 (find_package)
   cmake/pcl_find_qt.cmake:28 (find_package)
   CMakeLists.txt:432 (include)

 Configuring incomplete, errors occurred!
```

**Context**

Building pcl with Qt6.

**Expected behavior**
CMake should be able to successfully find and link to Qt6 without the error.

**Current Behavior**

If you switch from `WITH_QT=AUTO` to `WITH_QT=QT5` everything builds properly. Auto and QT6 produce the same error as seen above. 

**To Reproduce**
Ubuntu 24.04. 
Install Qt6 from apt. 
Try to run cmake for pcl with qt. 

**Your Environment (please complete the following information):**

 - OS: Ubuntu 24.04
 - Compiler: CMake 3.28.3
 - PCL Version: HEAD

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -423,6 +423,15 @@ if(WITH_VTK)
   endif()
 endif()
 
+# If VTK has found a Qt version, default to the same version
+if(VTK_QT_VERSION STREQUAL "5")
+  message(STATUS "VTK has found Qt5 - defaulting to same version.")
+  set(WITH_QT "QT5" CACHE STRING "Build QT Front-End (AUTO|YES|QT6|QT5|NO)")
+elseif(VTK_QT_VERSION STREQUAL "6")
+  message(STATUS "VTK has found Qt6 - defaulting to same version.")
+  set(WITH_QT "QT6" CACHE STRING "Build QT Front-End (AUTO|YES|QT6|QT5|NO)")
+endif()
+
 # VTK can depend on Qt and search for its required version, so search after so we can overwrite Qt5_FOUND/Qt6_FOUND if the version we require is not found.
 set(WITH_QT "AUTO" CACHE STRING "Build QT Front-End (AUTO|YES|QT6|QT5|NO)")
 set_property(CACHE WITH_QT PROPERTY STRINGS "AUTO" "YES" "QT6" "QT5" "NO")

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -430,11 +430,12 @@ if(VTK_QT_VERSION STREQUAL "5")
 elseif(VTK_QT_VERSION STREQUAL "6")
   message(STATUS "VTK has found Qt6 - defaulting to same version.")
   set(WITH_QT "QT6" CACHE STRING "Build QT Front-End (AUTO|YES|QT6|QT5|NO)")
+else()
+  # VTK can depend on Qt and search for its required version, so search after so we can overwrite Qt5_FOUND/Qt6_FOUND if the version we require is not found.
+  set(WITH_QT "AUTO" CACHE STRING "Build QT Front-End (AUTO|YES|QT6|QT5|NO)")
 endif()
-
-# VTK can depend on Qt and search for its required version, so search after so we can overwrite Qt5_FOUND/Qt6_FOUND if the version we require is not found.
-set(WITH_QT "AUTO" CACHE STRING "Build QT Front-End (AUTO|YES|QT6|QT5|NO)")
 set_property(CACHE WITH_QT PROPERTY STRINGS "AUTO" "YES" "QT6" "QT5" "NO")
+
 if(WITH_QT)
   include("${PCL_SOURCE_DIR}/cmake/pcl_find_qt.cmake")
 endif()

